### ISSUE TITLE
Discover 2.0 - possibility to expand multiple documents at the same time

### ISSUE THREAD

**Issue by maciejr-gain** [opensearch-project/OpenSearch-Dashboards#5555]
**Is your feature request related to a problem? Please describe.**

With all previous versions it was possible to expand as many Discover documents as users needed. It was heavily used by our users to compare few documents while debugging issues.
The new Discover unfortunately removed this possibility. If one document is expanded, in order to expand another one the prior one has to be closed.
This makes investigating by comparing three of four documents very difficult and time consuming.
Would it be possible to somehow bring back the old functionality where user is able to expand few documents at the same time?

**Describe the solution you'd like**

Possibility to expand multiple documents on the same page

**Comment by abbyhu2000** [comment]
@maciejr-gain Thanks for opening up the proposal!

@ashwin-pc Since expanding document detail has been changed to a flyout, it makes expanding multiple documents impossible. Should we add feature to allow this? @dagneyb @andrross 
<img width="1720" alt="Screenshot 2023-12-01 at 3 34 07 PM" src="https://github.com/opensearch-project/OpenSearch-Dashboards/assets/43937633/440a31c4-18eb-4bdf-849f-454156c5c359">

**Comment by ashwin-pc** [comment]
@kgcreative @shanilpa any thoughts on how to solve this with our new layout?

**Comment by kgcreative** [comment]
> @kgcreative @shanilpa any thoughts on how to solve this with our new layout?

I want to think about this more. Seems like expanding rows to view document details is a potentially desirable option, but I do think that the larger fly out does help with seeing more at once. We need to explore a few options.

First thing that comes to mind is making it so document details could be pinned and floated, that way you could have multiple floating windows that you could arrange and compare. I don't think we have a component for this today, but I could see it being useful for multiple use cases. Other options would be to allow for expanding or inspecting. Expanding would just expand the full json similar to legacy discover, but inspect would be the fully featured flyout with tabs and additional detail. I'm open to other thoughts as well

**Comment by kgcreative** [comment]
@shanilpa can you do a quick exploration of option 2, and/or propose a different alternative? Once we have that, then @ashwin-pc can scope and break it down

**Comment by shanilpa** [comment]
Hey @maciejr-gain - thanks for filing this issue.
I had a couple of clarifying questions around some of your users needs and workflows so I can better understand their workflows and recommend something that addresses them.

1. You mentioned in the previous versions of Discover users expanded multiple rows to compare against each other. Do you know how many they would typically open in a single view? 1-3, 4-7, or 7+?
2. Do you know what they are typically looking for when they expand the event?
3. Did the previous version of Discover that allowed for expanding events have missing functionality that your users wanted?

**Comment by maciejr-gain** [comment]
Hi @shanilpa,
let me try to reply to my best knowledge:
1. I am not sure exactly, we have quite a lot of users and I'm pretty sure the use scenarios vary between the teams. But my best guess would be that they would typically open 1-5 messages in a single view
2. In most cases they are trying to follow a stream of events, either within one application, or several applications that communicate with each other. They definitely rely on expanding the whole "@message" field, but they might be looking at other fields as well (most probably comparing "@timestamp", and several other, based on the application specific parsing rules).
3. We never got complaints about missing functionalities in the previous version of Discover
4. The one thing that I am sure they have been doing was adding additional filters based on the fields found in the Discover. Not sure if anything else.

I hope this helps, please ping me if you need anything else from me.

**Comment by maciejr-gain** [comment]
@shanilpa thank you for providing the options!
The first one looks definitely better and cleaner in my opinion. It gives choice, but when switched to inline the old look and feel are brought back.

The second options might work well for 2, maybe 3 expanded documents. If someone would like to expand 5 - I'm guessing it will not be readable anymore.

If I can vote, my vote is for option 1, no doubt.

**Comment by shanilpa** [comment]
@maciejr-gain thanks for the feedback!

@ashwin-pc @abbyhu2000 what would the engineering scope look like to get this change in for the next release?

**Comment by ashwin-pc** [comment]
cc: @ananzh

**Comment by michaelandrepearce** [comment]
+1 for option 1.

**Comment by kgcreative** [comment]
Your voice/vote as a user is even more important tbh

**Comment by damslo** [comment]
Hello, Option 1 looks great. Is there any chance to include it in 7.12.0? We have dozens users complaining about that.

**Comment by MadaniKK** [comment]
May I work on this? Thank you!

**Comment by ananzh** [comment]
@MadaniKK Thank you for the help!

**Comment by daihengbin724** [comment]
option 1 🎅

**Comment by AMoo-Miki** [comment]
> Hello, Option 1 looks great. Is there any chance to include it in 7.12.0? We have dozens users complaining about that.

OpenSearch Dashboards is in version 2; they have no. version 7.

**Comment by xodat2** [comment]
> > Hello, Option 1 looks great. Is there any chance to include it in 7.12.0? We have dozens users complaining about that.
> 
> OpenSearch Dashboards is in version 2; they have no. version 7.

Yes, I mean 2.12.0, anyway, I apparently it will be in 2.13

**Comment by kgcreative** [comment]
@wbeckler can we prioritize this for 2.12?

**Comment by abbyhu2000** [comment]
@MadaniKK How is the progress on this issue? Since we wish to target this for 2.12.0, do you have a timeline?

**Comment by MadaniKK** [comment]
Hi, I have not started on this issue because I was going to have a meeting with Anan to discuss this issue when she is back from her leave on 17th Jan. What should I do now?

**Comment by abbyhu2000** [comment]
@MadaniKK Since we wish to target this for 2.12, this issue will be prioritized by the team. Do you want to work on another issue that is not so urgent? For example, we can assign https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5616 to you?

**Comment by MadaniKK** [comment]
Yes sure! And it is ok I l still have some other issues to work on first. Thank you!

**Comment by henrik-ticket** [comment]
Option 1 please :heart:

**Comment by ankraj1** [comment]
Option 1 🙏

**Comment by nitin-kumar-ttn** [comment]
Option 1 please 🙏

**Comment by mz118** [comment]
Option 1 please 🙏

**Comment by ashwin-pc** [comment]
We are going with option 1. Thanks for all the feedback :)

**Comment by shanilpa** [comment]
@ashwin-pc @abbyhu2000 could we also work to contribute the changes to data grid back into OUI? This doesn't have to block the release of this enhancement but would be nice to have as an option for use elsewhere in the application.

**Comment by ananzh** [comment]
We restore legacy table in 2.12, which partially resolve the issue. Now only datagrid table missing this functionality. Update tag: remove bug and 2.12.
